gobblin-runtime/src/main/java/gobblin/runtime/util/JobStateToJsonConverter.java

```diff
@@ -87,6 +98,64 @@ public void convert(String jobName, String jobId, Writer writer)
       jsonWriter.close();
     }
   }
+  
+  /**
+   * Modify the configuration value of a given job instance.
+   */
+  public void modify(String jobName, String jobId, String key, String newValue) throws IOException {
+    List<? extends JobState> jobStates = this.jobStateStore.getAll(jobName, jobId + JOB_STATE_STORE_TABLE_SUFFIX);
+    if (jobStates.isEmpty()) {
+      LOGGER.warn(String.format("No job state found for job with name %s and id %s", jobName, jobId));
+      return;
+    }
+    JobState jobState = jobStates.get(0);
+
+    // Update the property in job state.
+    jobState.setProp(key, newValue);
+    
+    // Update the property in task state.
+    for (TaskState taskState: jobState.getTaskStates()) {
+      taskState.setProp(key, newValue);
+    }
+
+    this.replaceStateStore(jobName, jobId, jobState);
+  }
+
+  /**
+   * Modify the configuration value of the most recent job instance.
+   */
+  public void modify(String jobName, String key, String newValue) throws IOException {
+    this.modify(jobName, "current", key, newValue);
+  }
+  
+  /**
+   * Replace the original state file with a new state.
+   */
+  private void replaceStateStore(String jobName, String jobId, JobState newState) throws IOException {
+    Path storePath = new Path(storeUrl);
```

Comment:
You should try to use the `StateStore` API instead. Even though the current implementation is based on SequenceFile, it may change in the future.